src/test_utils.rs

```diff
@@ -102,9 +103,12 @@ impl TestContext {
         let (poison_sender, poison_receiver) = channel::bounded(1);
         async_std::task::spawn(async move {
             // Make sure that the test fails if there is a panic on this thread here:
+            let current_id = task::current().id();
             let orig_hook = panic::take_hook();
             panic::set_hook(Box::new(move |panic_info| {
-                poison_sender.try_send(panic_info.to_string()).ok();
+                if task::current().id() == current_id {
+                    poison_sender.try_send(panic_info.to_string()).ok();
+                }
```

Comment:
This prevents that lots of tests fail if one actually fails (tests run in parallel and before, the failing test sent a poison and then another test received it and also failed)